zilliqa-macros/src/test.rs

```diff
@@ -17,13 +17,18 @@ pub(crate) fn test_macro(_args: TokenStream, item: TokenStream) -> TokenStream {
     input.sig.ident = inner_name.clone();
 
     quote! {
+        use tracing::*;
+        use ntest::*;
+
         #[tokio::test(flavor = "multi_thread")]
+        #[timeout(300000_00)]
```

Comment:
IMO it doesn't make sense to time tests out based on wall-clock time, but instead based on the number of ticks of our simulated network. This will remain deterministic across machines, rather than having the situation where if a machine is too slow, its impossible to run our test suite.

Reply:
I will split this into a different PR for discussion.